src/Nethermind/Nethermind.Core/Transaction.cs

```diff
@@ -197,7 +197,11 @@ private void ClearPreHashInternal()
         /// </summary>
         public int GetLength(ITransactionSizeCalculator sizeCalculator, bool shouldCountBlobs)
         {
-            return _size ??= sizeCalculator.GetLength(this, shouldCountBlobs);
+            if (_size is not null) return _size.Value;
+
+            int size = sizeCalculator.GetLength(this, shouldCountBlobs);
+            if (shouldCountBlobs) _size = size;
+            return size;
```

Comment:
We shouldn't return cached `_size` when `shouldCountBlobs` is false
```suggestion
            return shouldCountBlobs 
              ? (_size ??= sizeCalculator.GetLength(this, true))
              : sizeCalculator.GetLength(this, false);
```

Reply:
Right, thanks for catching it!
Probably not worth to add `_sizeWithoutBlobs`, we are using it only once, during initial filtering